src/Microsoft.Azure.ServiceBus/Core/MessageReceiver.cs

```diff
@@ -916,13 +931,13 @@ public override void UnregisterPlugin(string serviceBusPluginName)
             }
         }
 
-        protected virtual async Task<IList<Message>> OnReceiveDeferredMessageAsync(IEnumerable<long> sequenceNumbers)
+        protected virtual async Task<IList<Message>> OnReceiveDeferredMessageAsync(long[] sequenceNumbers)
```

Comment:
I know i changed a public API here. but having an IEnumerable on a public api that is also in the fast path is problematic.

Given you dont know the producer of the IEnumerable, you dont know if it supports multiple iterations, or if it s a consume and discard approach. as such u need to play it safe and covert it to a local list or array, prior to doing multiple enumerations. You might want to consider making these types of APIs and IList, Icollection or an array

Reply:
Valid point :thumbsup: 

---
In reply to: [138780098](https://github.com/Azure/azure-service-bus-dotnet/pull/320#discussion_r138780098) [](ancestors = 138780098)

Reply:
@nemakam so given there are other `IEnumerable`s in the public API how would u like to handle those?

Reply:
@SimonCropp  Anything which is not part of interface, we could attempt at changing it. I don't want to change anything that's already part of interface. We should solve them by doing `.ToList()` and use them